### PR TITLE
core: debug chat and agent locks, update simplexmq

### PR DESCRIPTION
**Pull request by epoberezkin**

### CODE REVIEW

File: src/Simplex/Chat.hs [hunk 1]
```diff
@@ -281,7 +276,7 @@ processChatCommand = \case
     CTContactRequest -> pure $ chatCmdError "not implemented"
     CTContactConnection -> pure $ chatCmdError "not supported"
   APIGetChatItems _pagination -> pure $ chatCmdError "not implemented"
-  APISendMessage (ChatRef cType chatId) (ComposedMessage file_ quotedItemId_ mc) -> withUser $ \user@User {userId} -> withChatLock $ case cType of
+  APISendMessage (ChatRef cType chatId) (ComposedMessage file_ quotedItemId_ mc) -> withUser $ \user@User {userId} -> withChatLock "sendMessage" $ case cType of
```
Comment (jr-simplex) [inline src/Simplex/Chat.hs:279]: maybe use api method names in withChatLock?
```suggestion
  APISendMessage (ChatRef cType chatId) (ComposedMessage file_ quotedItemId_ mc) -> withUser $ \user@User {userId} -> withChatLock "APISendMessage" $ case cType of
```

File: src/Simplex/Chat.hs [hunk 2]
```diff
@@ -1472,7 +1471,7 @@ subscribeUserConnections agentBatchSubscribe user = do
         void . forkIO $ do
           threadDelay 1000000
           l <- asks chatLock
-          when (fileStatus == FSConnected) . unlessM (isFileActive fileId sndFiles) . withLock l $
+          when (fileStatus == FSConnected) . unlessM (isFileActive fileId sndFiles) . withLock l "sendFileChunk" $
```
Comment (jr-simplex) [inline src/Simplex/Chat.hs:1474]:
```suggestion
          when (fileStatus == FSConnected) . unlessM (isFileActive fileId sndFiles) . withLock l "subscribeUserConnections sendFileChunk" $
```

File: src/Simplex/Chat.hs [hunk 3]
```diff
@@ -1633,6 +1632,7 @@ processAgentMessage (Just user@User {userId, profile}) corrId agentConnId agentM
         MSG msgMeta _msgFlags msgBody -> do
           cmdId <- createAckCmd conn
           msg@RcvMessage {chatMsgEvent = ACME _ event} <- saveRcvMSG conn (ConnectionId connId) msgMeta msgBody cmdId
+          asks chatLock >>= \l -> void . atomically . swapTMVar l $ "directMessage " <> B.unpack (strEncode $ toCMEventTag event)
```
Comment (jr-simplex) [inline src/Simplex/Chat.hs:1635]: maybe do same in processGroupMessage

Comment (jr-simplex) [reply src/Simplex/Chat.hs:1635]: also maybe do it first before `cmdId <- createAckCmd conn`?

File: src/Simplex/Chat.hs [hunk 4]
```diff
@@ -1380,7 +1379,7 @@ agentSubscriber = do
   forever $ do
     (corrId, connId, msg) <- atomically $ readTBQueue q
     u <- readTVarIO =<< asks currentUser
-    withLock l . void . runExceptT $
+    withLock l ("subscriber " <> B.unpack (strEncode $ aCommandTag msg)) . void . runExceptT $
```
Comment (jr-simplex) [inline src/Simplex/Chat.hs:1382]: log connId and corrId?
```suggestion
    withLock l ("subscriber connId=" <> B.unpack (strEncode connId) <> " corrId=" <> B.unpack (strEncode corrId) <> " aCmdTag=" <> B.unpack (strEncode $ aCommandTag msg)) . void . runExceptT $
```

File: src/Simplex/Chat.hs (hunk 1 above)
Comment (epoberezkin) [reply src/Simplex/Chat.hs:279]: I was thinking to use what's closer to chat console

File: src/Simplex/Chat.hs (hunk 3 above)
Comment (epoberezkin) [reply src/Simplex/Chat.hs:1635]: added to group; there is no `event` before createAckCmd

**Pull request merged**